packages/nextjs/app/debug/_components/contract/__test__/utilsContract.test.tsx

```diff
@@ -0,0 +1,118 @@
+import { describe, expect, it } from "vitest";
+import { getArgsAsStringInputFromForm } from "../utilsContract";
+
+describe("utilsContract", () => {
+  it("should parse integer form correctly", () => {
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::integer::u8": '10' }, false, true)).toEqual([10]);
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::integer::u16": '10' }, false, true)).toEqual([10]);
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::integer::u32": '10' }, false, true)).toEqual([10]);
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::integer::u64": '10' }, false, true)).toEqual([10]);
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::integer::u128": '10' }, false, true)).toEqual([10]);
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::integer::u256": '10' }, false, true)).toEqual([10]);
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::felt252": '10' }, false, true)).toEqual([10]);
+
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::integer::i8": '10' }, false, true)).toEqual([10]);
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::integer::i16": '10' }, false, true)).toEqual([10]);
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::integer::i32": '10' }, false, true)).toEqual([10]);
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::integer::i64": '10' }, false, true)).toEqual([10]);
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::integer::i128": '10' }, false, true)).toEqual([10]);
+  });
+
+  it("should parse bool form correctly", () => {
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::bool": 'true' }, false, true)).toEqual([true]);
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::bool": '1' }, false, true)).toEqual([true]);
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::bool": '0x1' }, false, true)).toEqual([true]);
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::bool": '0x01' }, false, true)).toEqual([true]);
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::bool": '0x001' }, false, true)).toEqual([true]);
+
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::bool": 'false' }, false, true)).toEqual([false]);
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::bool": '0' }, false, true)).toEqual([false]);
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::bool": '0x0' }, false, true)).toEqual([false]);
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::bool": '0x00' }, false, true)).toEqual([false]);
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::bool": '0x000' }, false, true)).toEqual([false]);
+  });
+
```

Comment:
would request one for tuples as well , other than that we're good

Reply:
sure, will add it